app/views/signIn.scala.html

```diff
@@ -28,18 +26,21 @@
             <p class="not-a-member">Not a member? <a href="@routes.ApplicationController.signUp">Sign up now</a></p>
         </div>
 
-        @defining(env.providers.values.filter(_.isInstanceOf[SocialProvider])) { providers =>
-            @if(!providers.isEmpty) {
-                <div class="social-providers">
-                    <p>Or use your existing account on one of the following services to sign in:</p>
-                    <div>
-                    @for(p <- providers if p.id != "clef") {
-                        <a href="@routes.SocialAuthController.authenticate(p.id)" class="provider @p.id"><img src="@routes.Assets.at(s"images/providers/${p.id}.png")" style="width:64px;height:64px;"></a>
-                    }
-                    </div>
-                    <script type="text/javascript" src="https://clef.io/v3/clef.js" class="clef-button" data-app-id="@Play.configuration.getString("silhouette.clef.clientID").getOrElse("")" data-color="blue" data-style="flat" data-redirect-url="@Play.configuration.getString("silhouette.clef.redirectURL").getOrElse("")" data-type="login"></script>
+        @if(socialProviders.providers.nonEmpty) {
+            <div class="social-providers">
+                <p>Or use your existing account on one of the following services to sign in:</p>
+                <div>
+                @for(p <- socialProviders.providers if p.id != "clef") {
+                    <a href="@routes.SocialAuthController.authenticate(p.id)" class="provider @p.id"><img src="@routes.Assets.at(s"images/providers/${p.id}.png")" style="width:64px;height:64px;"></a>
+                }
                 </div>
-            }
+                @defining(socialProviders.get[ClefProvider]) { clef =>
+                    @if(clef.isDefined) {
+                        <script type="text/javascript" src="https://clef.io/v3/clef.js" class="clef-button" data-app-id="@clef.get.settings.clientID" data-color="blue" data-style="flat" data-redirect-url="@clef.get.settings.redirectURL" data-type="login"></script>
+                    }
```

Comment:
It would be nicer to move this into the for loop and use pattern matching so that order can be maintained
```scala
     @for(p <- socialProviders.providers) {
        case clef:ClefProvider =>
          <script type="text/javascript" src="https://clef.io/v3/clef.js" class="clef-button" data-app-id="@clef.get.settings.clientID" data-color="blue" data-style="flat" data-redirect-url="@clef.get.settings.redirectURL" data-type="login"></script>
        case _ =>
          <a href="@routes.SocialAuthController.authenticate(p.id)" class="provider @p.id"><img src="@routes.Assets.at(s"images/providers/${p.id}.png")" style="width:64px;height:64px;"></a>
      }
```